code/datums/components/uplink.dm

```diff
@@ -254,6 +248,12 @@
 
 	var/mob/user = arguments[2]
 	owner = "[user.key]"
+	if(owner)
+		LAZYINITLIST(GLOB.uplink_purchase_logs_by_key)
+		if(GLOB.uplink_purchase_logs_by_key[owner])
+			purchase_log = GLOB.uplink_purchase_logs_by_key[owner]
+		else
+			purchase_log = new(owner, src)
```

Comment:
If you move this code to `implanting`, won't this mean that uplinks that are embedded in radios or PDAs won't ever create a purchase log?

Reply:
Hmm, actually i think after I fixed a bug causing owner to not propagate I think just testing for if(owner) should be sufficient so I can move it back up